### PR TITLE
Cleanup unknown props

### PR DESCRIPTION
**Pull request by agudulin**
React 15 introduced these warning messages about unknown props injected into DOM elements.
We want to clean them up.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,20 +1,28 @@
...
+import * as pluginOriginalClass from './makers/makeOriginalClass'
+import * as pluginStates from './makers/makeStates'
 
-const basicMakers = [makeBlock, makeOriginalClass, makeModifiers, makeStates]
+const basicPlugins = [pluginBlock, pluginModifiers, pluginOriginalClass, pluginStates]
+
+const filterProps = (props, unknownProps) => {
+  const keys = Object.keys(props).filter(prop => unknownProps.indexOf(prop) < 0)
+
+  return keys.map(key => props[key])
```
Comment (robinpokorny) [inline src/index.js]: Should not this return an object?

File: src/index.js [hunk 2]
```diff
@@ -1,20 +1,28 @@
...
-import makeStates from './makers/makeStates'
+import * as pluginBlock from './makers/makeBlock'
+import * as pluginModifiers from './makers/makeModifiers'
+import * as pluginOriginalClass from './makers/makeOriginalClass'
+import * as pluginStates from './makers/makeStates'
 
-const basicMakers = [makeBlock, makeOriginalClass, makeModifiers, makeStates]
+const basicPlugins = [pluginBlock, pluginModifiers, pluginOriginalClass, pluginStates]
+
+const filterProps = (props, unknownProps) => {
```
Comment (robinpokorny) [inline src/index.js]: I do not like the name `unknownProps`.

Comment (robinpokorny) [inline src/index.js]: From Babel:

``` js
function _objectWithoutProperties(obj, keys) {
  var target = {};
  for (var i in obj) {
    if (keys.indexOf(i) >= 0) continue;
    if (!Object.prototype.hasOwnProperty.call(obj, i)) continue;
    target[i] = obj[i]; }
  return target;
}

```

Comment (robinpokorny) [inline src/index.js]: Or you can use https://lodash.com/docs/4.16.2#omit

File: src/index.js [hunk 3]
```diff
@@ -1,20 +1,28 @@
...
+  const { extraPlugins = [] } = options
+  const { element, ...restProps } = props
   const blockName = block + (element ? `__${element}` : '')
 
-  const makers = basicMakers
-    .concat(extraMakers)
-    .map((maker) => maker(blockName, props))
+  const plugins = basicPlugins.concat(extraPlugins)
+
+  const makers = plugins.map(plugin => plugin.maker(blockName, props))
```
Comment (robinpokorny) [inline src/index.js]: I would differentiate pluck and call

File: src/index.js [hunk 4]
```diff
@@ -1,20 +1,28 @@
...
+const filterProps = (props, unknownProps) => {
+  const keys = Object.keys(props).filter(prop => unknownProps.indexOf(prop) < 0)
+
+  return keys.map(key => props[key])
+}
 
 export default (block, options = {}) => (props) => {
-  const { extraMakers = [] } = options
-  const { element } = props
+  const { extraPlugins = [] } = options
```
Comment (robinpokorny) [inline src/index.js]: Just `plugins`?

**Review by robinpokorny — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 2 above)
Comment (agudulin) [reply src/index.js]: `s/unknownProps/propsToRemove`?

**Comment by agudulin** [comment]
@robinpokorny have another look?

File: src/index.js [hunk 5]
```diff
@@ -1,20 +1,26 @@
 import cx from 'classnames'
 
-import makeBlock from './makers/makeBlock'
-import makeModifiers from './makers/makeModifiers'
-import makeOriginalClass from './makers/makeOriginalClass'
-import makeStates from './makers/makeStates'
+import * as pluginBlock from './makers/makeBlock'
```
Comment (robinpokorny) [inline src/index.js]: A small one, should we rename the folder to `plugins`?

**Review by robinpokorny — APPROVED** [review]

File: src/index.js (hunk 5 above)
Comment (agudulin) [reply src/index.js]: oh, we should, thanks 👍

**Comment by robinpokorny** [comment]
Looks good! 👍

**Pull request merged**